scripts/mitochondria_m2_wdl/AlignAndCall.wdl

```diff
@@ -169,93 +202,101 @@ workflow AlignAndCall {
     File mt_aligned_shifted_bai = AlignToShiftedMt.mt_aligned_bai
-    File out_vcf = Filter.filtered_vcf
-    File out_vcf_index = Filter.filtered_vcf_idx
+    File out_vcf = FilterLowHetSites.final_filtered_vcf
+    File out_vcf_index = FilterLowHetSites.final_filtered_vcf_idx
+    File input_vcf_for_haplochecker = SplitMultiAllelicsAndRemoveNonPassSites.vcf_for_haplochecker
     File duplicate_metrics = AlignToMt.duplicate_metrics
     File coverage_metrics = CollectWgsMetrics.metrics
     File theoretical_sensitivity_metrics = CollectWgsMetrics.theoretical_sensitivity
     File contamination_metrics = GetContamination.contamination_file
     Int mean_coverage = CollectWgsMetrics.mean_coverage
     String major_haplogroup = GetContamination.major_hg
-    Float contamination = GetContamination.minor_level
+    Float contamination = FilterContamination.contamination
   }
 }
 
+
 task GetContamination {
   input {
-    File input_bam
-    File input_bam_index
-    File ref_fasta
-    File ref_fasta_index
-    Int qual = 20
-    Int map_qual = 30
-    Float vaf = 0.01
+    File input_vcf
+      # runtime
+    Int? preemptible_tries}
 
-    # runtime
-    Int? preemptible_tries
-  }
-
-  String basename = basename(input_bam, ".bam")
-  Float ref_size = size(ref_fasta, "GB") + size(ref_fasta_index, "GB")
-  Int disk_size = ceil(size(input_bam, "GB") + ref_size) + 20
+  Int disk_size = ceil(size(input_vcf, "GB")) + 20
 
   meta {
-    description: "Uses Haplochecker to estimate levels of contamination in mitochondria"
+    description: "Uses new Haplochecker to estimate levels of contamination in mitochondria"
   }
   parameter_meta {
-    input_bam: "Bam aligned to chrM"
-    ref_fasta: "chrM reference"
+    input_vcf: "Filtered and split multi-allelic sites VCF for mitochondria"
   }
-  command {
+  command <<<
   set -e
-
-  java -jar /usr/mtdnaserver/mitolib.jar haplochecker \
-    --in ~{input_bam} \
-    --ref ~{ref_fasta} \
-    --out haplochecker_out \
-    --QUAL ~{qual} \
-    --MAPQ ~{map_qual} \
-    --VAF ~{vaf}
-
-python3 <<CODE
-
-import csv
-
-with open("haplochecker_out/~{basename}.contamination.txt") as output:
-    reader = csv.DictReader(output, delimiter='\t')
-    for row in reader:
-        print(row["MajorHG"], file=open("major_hg.txt", 'w'))
-        print(row["MajorLevel"], file=open("major_level.txt", 'w'))
-        print(row["MinorHG"], file=open("minor_hg.txt", 'w'))
-        print(row["MinorLevel"], file=open("minor_level.txt", 'w'))
-CODE
-  }
+  PARENT_DIR="$(dirname "~{input_vcf}")"
+  java -jar /usr/mtdnaserver/haplocheckCLI.jar "${PARENT_DIR}"
+
+  sed 's/\"//g' output > output-noquotes
+  grep -v "SampleID" output-noquotes > output-data
+  awk '{print $2}' output-data > contamination.txt
+  awk '{print $6}' output-data > major_hg.txt
+  awk '{print $8}' output-data > minor_hg.txt
+  awk '{print $14}' output-data > mean_het_major.txt
+  awk '{print $15}' output-data > mean_het_minor.txt
```

Comment:
Do you think it's worth adding any sanity checks here to make sure this doesn't break in the future since it's relying on column order to extract the values?